spring-security-modules/spring-security-web-rest/src/main/java/com/baeldung/security/SecurityJavaConfig.java

```diff
@@ -53,36 +55,17 @@ public InMemoryUserDetailsManager userDetailsService() {
 
     @Bean
     public SecurityFilterChain filterChain(HttpSecurity http) throws Exception {
-        http.csrf()
-            .disable()
-            .authorizeRequests()
-            .and()
-            .exceptionHandling()
-            .accessDeniedHandler(accessDeniedHandler)
-            .authenticationEntryPoint(restAuthenticationEntryPoint)
-            .and()
-            .authorizeRequests()
-            .antMatchers("/api/csrfAttacker*")
-            .permitAll()
-            .antMatchers("/api/customer/**")
-            .permitAll()
-            .antMatchers("/api/foos/**")
-            .authenticated()
-            .antMatchers("/api/async/**")
-            .permitAll()
-            .antMatchers("/api/admin/**")
-            .hasRole("ADMIN")
-            .and()
-            .formLogin()
-            .successHandler(mySuccessHandler)
-            .failureHandler(myFailureHandler)
-            .and()
-            .httpBasic()
-            .and()
-            .logout();
+        http.authorizeRequests(authorizeRequests -> authorizeRequests.requestMatchers("/api/csrfAttacker*").permitAll()
+            .requestMatchers("/api/customer/**").permitAll()
+            .requestMatchers("/api/foos/**").authenticated()
+            .requestMatchers("/api/async/**").permitAll()
+            .requestMatchers("/api/admin/**").hasRole("ADMIN"))
+            .formLogin(formLogin -> formLogin.successHandler(mySuccessHandler).failureHandler(myFailureHandler))
+            .httpBasic(withDefaults())
+            .logout(logout -> logout.permitAll());
```

Comment:
Please use method reference instead of lambda.

Reply:
@alvin577austria Could you kindly provide any specific reasons for this? I followed the code practice outlined in the documentation provided by Spring.

Reply:
I intended the comment for line 63, but it seems using method reference is not possible. Ignore this.